### PR TITLE
Java cleanup

### PR DESCRIPTION
**Pull request by jsoref**
This is for discussion.

Notes:
* Most commits were created by IntelliJ (the descriptions are mine because I lost the details)
* Individual commits are mostly self contained
* In general, most commits should not change program behavior at all

This is generally a subset of https://github.com/jsoref/jenkins/commits/java-cleanup-full
 

### Proposed changelog entries

* `Internal:` Internal Java code cleanup

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

File: core/src/main/java/hudson/model/UpdateSite.java [hunk 1]
```diff
@@ -283,12 +283,12 @@ protected JSONSignatureValidator getJsonSignatureValidator(@CheckForNull String
     /**
      * Returns true if it's time for us to check for new version.
      */
-    public boolean isDue() {
+    synchronized public boolean isDue() {
```
Comment (Wadeck) [inline core/src/main/java/hudson/model/UpdateSite.java]: Adding `synchronized` using an automatic tool, without further explanation, I think that's a bad idea.

File: core/src/main/java/hudson/ClassicPluginStrategy.java [hunk 2]
```diff
@@ -137,6 +137,7 @@ private static Manifest loadLinkedManifest(File archive) throws IOException {
                 } catch (InvalidPathException e) {
                     throw new IOException(e);
                 }
+                //noinspection StatementWithEmptyBody
```
Comment (Wadeck) [inline core/src/main/java/hudson/ClassicPluginStrategy.java]: I am not sure we really want to have such instruction in the code. That's very specific to IntelliJ, compared to common annotation like `@SuppressWarning`. 
Opening the door for comment/annotation specific to all the popular tools is perhaps really desired.

For the moment we only have 2 of them, from you, in the core.

**Review by Wadeck — COMMENTED** [review]
Concerning the added synchronization, I would require to have more information/explanation.

For the IntelliJ specific comment, I just want to highlight that part to open the discussion as it was not the case during #3909.

File: core/src/main/java/hudson/model/UpdateSite.java (hunk 1 above)
Comment (jsoref) [reply core/src/main/java/hudson/model/UpdateSite.java]: I'll drop from this PR…

File: core/src/main/java/hudson/ClassicPluginStrategy.java (hunk 2 above)
Comment (jsoref) [reply core/src/main/java/hudson/ClassicPluginStrategy.java]: I'll drop from this PR…

File: core/src/main/java/hudson/FilePath.java [hunk 3]
```diff
@@ -335,7 +335,8 @@ public static String normalize(@Nonnull String path) {
                 tokens.add(path.substring(s, i));
                 s = i;
                 // Skip any extra separator chars
-                while (++i < end && ((c = path.charAt(i)) == '/' || c == '\\')) { }
+                //noinspection StatementWithEmptyBody
+                while (++i < end && ((c = path.charAt(i)) == '/' || c == '\\'));
```
Comment (daniel-beck) [inline core/src/main/java/hudson/FilePath.java]: Easier to read as in https://github.com/jenkinsci/jenkins/pull/4020/files#diff-bf28198e17c0bde7c545a51b280856b9R132

```suggestion
                while (++i < end && ((c = path.charAt(i)) == '/' || c == '\\'))
                    ;
```

File: core/src/main/java/hudson/util/ChunkedInputStream.java [hunk 4]
```diff
@@ -344,7 +344,7 @@ public void close() throws IOException {
     static void exhaustInputStream(InputStream inStream) throws IOException {
         // read and discard the remainder of the message
         byte[] buffer = new byte[1024];
-        while (inStream.read(buffer) >= 0) {
-        }
+        //noinspection StatementWithEmptyBody
+        while (inStream.read(buffer) >= 0);
```
Comment (daniel-beck) [inline core/src/main/java/hudson/util/ChunkedInputStream.java]:
```suggestion
        while (inStream.read(buffer) >= 0)
            ;
```

File: core/src/main/java/hudson/ClassicPluginStrategy.java (hunk 2 above)
Comment (daniel-beck) [reply core/src/main/java/hudson/ClassicPluginStrategy.java]: FWIW I think an empty block with a comment in it is a legitimate tool to improve code readability and any tool should be configurable to ignore that with such a check. Perhaps a feature request to IntelliJ, or wherever this is from?

**Comment by daniel-beck** [comment]
Forgot to comment: Largely LGTM with the exception of the synchronized change Wadeck commented on, and (less important) my suggested changes. Close to done and mergeable IMO.

File: core/src/main/java/hudson/ClassicPluginStrategy.java (hunk 2 above)
Comment (jsoref) [reply core/src/main/java/hudson/ClassicPluginStrategy.java]: This is from IntelliJ:
image
image

I think based on the comment that there isn't a current standard `@Suppress` annotation which is why IntelliJ is using a comment, in general it uses `@...` things...

Comment (jsoref) [reply core/src/main/java/hudson/ClassicPluginStrategy.java]: So, the problem is that the `@SuppressWarnings` are a function of the compiler (of which there are a couple), and there isn't one for this thing, so it seems that IDEs are using non-standard extensions.

**Comment by jsoref** [comment]
@Wadeck: I've dropped the synchronized commit.

**Review by Wadeck — APPROVED** [review]
:bee: thank you!

**Comment by jsoref** [comment]
@daniel-beck: I've pulled in your suggestions and dropped the synchronized commit.

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/java/hudson/FilePath.java [hunk 5]
```diff
@@ -335,7 +335,9 @@ public static String normalize(@Nonnull String path) {
                 tokens.add(path.substring(s, i));
                 s = i;
                 // Skip any extra separator chars
-                while (++i < end && ((c = path.charAt(i)) == '/' || c == '\\')) { }
+                //noinspection StatementWithEmptyBody
+                while (++i < end && ((c = path.charAt(i)) == '/' || c == '\\'))
+                    ;
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/FilePath.java]:
```
{
  // noop
}
```

would be better. Anywa, I prefer the original code more

**Review by oleg-nenashev — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
i think this is ready to go. Will merge tomorrow if no negative feedback

**Pull request merged**